.circleci/config.yml

```diff
@@ -0,0 +1,43 @@
+version: 2
+jobs:
+  checkout_cuberite:
+    working_directory: ~/cuberite
+    docker:
+    - image: peterbell10/cuberite-circleci:0.0.3
+    steps:
+    - checkout
+    - run: git submodule update --init
+    - save_cache:
+        key: "cuberite-{{ .Environment.CIRCLE_SHA1 }}"
+        paths:
+        - ~/cuberite
+
+  check_basic_style:
+    working_directory: ~/cuberite/src
+    docker:
+    - image: peterbell10/cuberite-circleci:0.0.3
+    steps:
+    - restore_cache: { keys: [ "cuberite-{{ .Environment.CIRCLE_SHA1 }}" ] }
+    - run: find . -name \*.cpp -or -name \*.h > AllFiles.lst
```

Comment:
It is already created by [src/CMakeLists.txt](https://github.com/cuberite/cuberite/blob/master/src/CMakeLists.txt#L285).
But maybe it would be a good additional test to check, that the `find` output is identical to the one from CMake (in case it missed a file). I could create an Issue, if this new test doesn't fit into this PR.

Reply:
`cmake` doesn't run on this job.